NEWS.md

```diff
@@ -1,203 +1,117 @@
-   ([#3843](https://github.com/syslog-ng/syslog-ng/pull/3843))
-
-## Notes to developers
-
- * plugins: we have made it easier to implement filter plugins
-
-   An example can be found under `modules/rate-limit-filter`.
-   ([#3866](https://github.com/syslog-ng/syslog-ng/pull/3866))
- * dev-utils: various fixes for the plugin skeleton generator script
-   ([#3866](https://github.com/syslog-ng/syslog-ng/pull/3866))
+  * `http()` and other threaded destinations: fix `$SEQNUM` processing so that
+    only local messages get an associated `$SEQNUM`, just like normal
+    `syslog()`-like destinations.  This avoids a [meta sequenceId="XXX"] SD-PARAM
+    being added to `$SDATA` for non-local messages.
+    ([#3928](https://github.com/syslog-ng/syslog-ng/pull/3928))
+  * `grouping-by()`: fix `grouping-by()` use through parser references.
+    Originally if a grouping-by() was part of a named parser statement and was
+    referenced from multiple log statements, only the first `grouping-by()`
+    instance behaved properly, 2nd and subsequent references were ignoring all
+    configuration options and have reverted using defaults instead.
+    ([#3957](https://github.com/syslog-ng/syslog-ng/pull/3957))
+  * `db-parser()`: similarly to `grouping-by()`, `db-parser()` also had issues
+    propagating some of its options to 2nd and subsequent references of a parser
+    statement. This includes `drop-unmatched()`, `program-template()` and
+    `template()` options.
+    ([#3957](https://github.com/syslog-ng/syslog-ng/pull/3957))
+  * `match(), subst() and regexp-parser()`: fixed storing of numbered
+    (e.g.  $1,$2, $3 and so on) and named capture groups in regular expressions
+    in case the input of the regexp is the same as one of the match variables being
+    stored. In some cases the output of the regexp was clobbered and an invalid
+    value stored.
+    ([#3948](https://github.com/syslog-ng/syslog-ng/pull/3948))
+  * fix `threaded(no)` related crash: if threaded mode is disabled for
+    asynchronous sources and destinations (all syslog-like drivers such as
+    tcp/udp/syslog/network qualify), a use-after-free condition can happen due
+    to a reference counting bug in the non-threaded code path.  The
+    `threaded(yes)` setting has been the default since 3.6.1 so if you are using
+    a more recent version, you are most probably unaffected.  If you are using
+    `threaded(no)` a use-after-free condition happens as the connection closes.
+    The problem is more likely to surface on 32 bit platforms due to pointer
+    sizes and struct layouts where this causes a NULL pointer dereference.
+    ([#3997](https://github.com/syslog-ng/syslog-ng/pull/3997))
+
+## Packaging
+
+  * debian: fix logrotate file not doing the file rotation. (The path and command was invalid.)
+    ([#4031](https://github.com/syslog-ng/syslog-ng/pull/4031))
+  * `dbld`: add support for Fedora 35 in favour of Fedora 33
```

Comment:
Let's move this under other changes. The packaging section is about rpm/deb-related changed that might be interesting for package maintainers.

I would also consider moving the logrotate fix to bugfixes for the same reason, but I'm happy with its current location as well.